### ISSUE TITLE
Configurable product prices are converted twice in PriceManager ProductWithChildren

### ISSUE THREAD

**Issue by rakibabu** [algolia/algoliasearch-magento-2#578]
**Do you want to request a *feature* or report a *bug*?**
Bug

**Bug: What is the current behavior?**

Configurable product prices are converted twice in `vendor/algolia/algoliasearch-magento-2/Helper/Entity/Product/PriceManager/ProductWithChildren.php` in the method handleZeroDefaultPrice. Simple products are converted as excepted. When commenting out this lines 100 untill 102: 

```
if ($currencyCode !== $this->baseCurrencyCode) {
    $min = $this->convertPrice($min, $currencyCode);
}
```

The prices are converted as expected. 

Before commenting out in Algolia dashboard:

![conversion-twice](https://user-images.githubusercontent.com/14089150/46068503-97654480-c179-11e8-9cec-37ce486a4749.jpg)

After commenting out in the Algolia dashboard:

![conversion-once](https://user-images.githubusercontent.com/14089150/46068571-c1b70200-c179-11e8-8e78-f7ce7f72a8bb.jpg)

**Bug: What is the expected behavior?**

Expected is that prices are not converted twice, only once. 

**Bug: What is the proposed solution?**

When commenting out line 100 until 102 of the`vendor/algolia/algoliasearch-magento-2/Helper/Entity/Product/PriceManager/ProductWithChildren.php` of the handleZeroDefaultPrice method  everything is working as excepted. Maybe set a variable if the prices are already converted, so this method not converting the prices twice?

**What is the version of Magento and of Algolia extension you are using? Always use the latest version of the extension one before opening a bug issue.**

This bug is on Magento 2.2.5 & 2.2.6
Algolia version: 1.8.1

**Comment by JanPetr** [comment]
Hello @rakibabu,

thank you for the report! We'll take a look and come back to you soon.

**Comment by JanPetr** [comment]
Hello @rakibabu,

the fix is released in 1.8.2 :)

**Comment by PardeepGrover** [comment]
Hi 
i am using 1.9.0 version 

below error is coming on re-indexing.

Array must contain at least one element in /home/uprj10551/www/vendor/algolia/algoliasearch-magento-2/Helper/Entity/Product/PriceManager/Configurable.php on line 22
